File: src/main/java/net/dontdrinkandroot/gitki/service/lock/LockService.java
package net.dontdrinkandroot.gitki.service.lock;

import net.dontdrinkandroot.gitki.model.FilePath;
import net.dontdrinkandroot.gitki.model.GitUser;
import net.dontdrinkandroot.gitki.model.LockInfo;

import java.util.List;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public interface LockService
{
    LockInfo lock(FilePath filePath, GitUser user) throws LockedException;

    void release(FilePath filePath, GitUser user) throws LockedException;

    /**
     * Forces the release of the Lock for the given FilePath.
     *
     * @param filePath The FilePath to unlock.
     */
    void forceRelease(FilePath filePath);

    LockInfo check(FilePath filePath);

    List<LockInfo> list();

    /**
     * Clears all locks.
     */
    void clear();
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/service/lock/LockService.kt
package net.dontdrinkandroot.gitki.service.lock

import net.dontdrinkandroot.gitki.model.FilePath
import net.dontdrinkandroot.gitki.model.GitUser
import net.dontdrinkandroot.gitki.model.LockInfo

interface LockService {

    @Throws(LockedException::class)
    fun lock(filePath: FilePath, user: GitUser): LockInfo

    @Throws(LockedException::class)
    fun release(filePath: FilePath, user: GitUser)

    /**
     * Forces the release of the Lock for the given FilePath.
     *
     * @param filePath The FilePath to unlock.
     */
    fun forceRelease(filePath: FilePath)

    fun check(filePath: FilePath): LockInfo?

    fun list(): List<LockInfo>

    /**
     * Clears all locks.
     */
    fun clear()
}